src/jit/optcse.cpp

```diff
@@ -2073,15 +2088,15 @@ class CSE_Heuristic
             /* Advance to the next node in the list */
             lst = lst->tslNext;
 
-            // Assert if we used DEBUG_DESTROY_NODE on this CSE exp
-            assert(exp->gtOper != GT_COUNT);
-
-            /* Ignore the node if it's not been marked as a CSE */
+            // Ignore this node if the gtCSEnum value has been cleared
```

Comment:
I assume this diff and the next are the actual fix and the remaining diffs are just incidental fixes?

Might be nice to have a reminder here of how it is `gcCSEnum` can end up getting cleared.

Reply:
We can clear a CSE def when we encounter two defs with different exception sets.
It is cleared in optValuenumCSE_Availablity.  See:

`  JITDUMP(" Abandoned - CSE candidate has defs with different exception sets!\n");`